src/main/java/fr/anatom3000/gwwhit/materials/CustomOre.java

```diff
@@ -107,10 +158,229 @@ public void onInitialize() {
+                            .key("S", JIngredient.ingredient().item(Items.STICK)),
+                    JResult.result(String.format("%s:%s_axe", MOD_ID, name.toLowerCase()))
+            ));
+            addRecipe("hoe", JRecipe.shaped(
+                    JPattern.pattern("MM", "S ", "S "),
+                    JKeys.keys()
+                            .key("M", JIngredient.ingredient().item(materialId.toString()))
+                            .key("S", JIngredient.ingredient().item(Items.STICK)),
+                    JResult.result(String.format("%s:%s_hoe", MOD_ID, name.toLowerCase()))
+            ));
+            addRecipe("pickaxe", JRecipe.shaped(
+                    JPattern.pattern("MMM", " S ", " S "),
+                    JKeys.keys()
+                            .key("M", JIngredient.ingredient().item(materialId.toString()))
+                            .key("S", JIngredient.ingredient().item(Items.STICK)),
+                    JResult.result(String.format("%s:%s_pickaxe", MOD_ID, name.toLowerCase()))
+            ));
+            addRecipe("shovel", JRecipe.shaped(
+                    JPattern.pattern("M ", "S ", "S "),
+                    JKeys.keys()
+                            .key("M", JIngredient.ingredient().item(materialId.toString()))
+                            .key("S", JIngredient.ingredient().item(Items.STICK)),
+                    JResult.result(String.format("%s:%s_shovel", MOD_ID, name.toLowerCase()))
+            ));
+        }
+        if (hasSword) {
+            addRecipe("sword", JRecipe.shaped(
+                    JPattern.pattern("M ", "M ", "S "),
+                    JKeys.keys()
+                            .key("M", JIngredient.ingredient().item(materialId.toString()))
+                            .key("S", JIngredient.ingredient().item(Items.STICK)),
+                    JResult.result(String.format("%s:%s_sword", MOD_ID, name.toLowerCase()))
+            ));
+        }
+        addRecipe("ore_smelting", JRecipe.smelting(
+                JIngredient.ingredient().item(oreId.toString()),
+                JResult.item(material)
+        ).cookingTime(rnd.nextInt(400)).experience(rnd.nextFloat() * 5));
+        if (type == Type.INGOT) {
+            addRecipe("ore_blasting", JRecipe.blasting(
+                    JIngredient.ingredient().item(oreId.toString()),
+                    JResult.item(material)
+            ).cookingTime(rnd.nextInt(200)).experience(rnd.nextFloat()));
+        }
+        addRecipe("block", JRecipe.shaped(
+                JPattern.pattern("MMM", "MMM", "MMM"),
+                JKeys.keys().key("M", JIngredient.ingredient().item(materialId.toString())),
+                JResult.result(blockId.toString())
+        ));
+        addRecipe(type.toString(), JRecipe.shapeless(
+                JIngredients.ingredients().add(JIngredient.ingredient().item(blockId.toString())),
+                JResult.itemStack(material, 9)
+        ));
+        if (rnd.nextInt(10) == 0) {
```

Comment:
Maybe pass the dye color as an argument? The dye recipes weren't random, but based on ore color.

Reply:
This is probably a good idea, that argument could also be used if someone wants to try to implement texture generation (the only non-procedural piece apart from the lang file currently)

Reply:
i mean when you manage to procedurally generate textures you wouldn't need to pass anything to the class except a seed and just generate the rest

Reply:
Done